aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/LastSyncMetadata.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.datastore.syncengine;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.core.util.ObjectsCompat;
+
+import com.amplifyframework.core.model.Model;
+import com.amplifyframework.core.model.annotations.ModelField;
+
+import java.util.Objects;
+import java.util.UUID;
+
+/**
+ * Metadata about the last time that a model class was sync'd with AppSync backend.
+ * This metadata is persisted locally as a system model. This metadata is inspected
+ * whenever the Sync Engine starts up. The system consider the value of
+ * {@link LastSyncMetadata#getLastSyncTime()} to decide whether or not it should
+ * perform a "Base Sync" or a "Delta Sync".
+ */
+public final class LastSyncMetadata implements Model {
```

Comment:
There's probably a good reason for this I'm just missing here but was there a reason you decided to go with a new model for this metadata vs. supplementing the existing metadata object with a lastSyncTime field?

Reply:
The `ModelMetadata` deals with metadata for a single _instance_ of a model.

The `LastSyncMetadata` deals with metadata for the model _type_, inclusive of any/all instances there-of.

However, in the biggest picture, I agree with what you're saying, and I do think we could probably simplify our metadata / system models a bit.

[Here are the ones we have right now, complete with (decent) doco](https://github.com/aws-amplify/amplify-android/blob/master/aws-datastore/src/main/java/com/amplifyframework/datastore/storage/SystemModelsProviderFactory.java#L39-L51).